chore(blog): small tweaks to schema customization blog post (#14090)
gatsbyjs/gatsby: docs/blog/2019-05-17-improvements-to-schema-customization/index.md

  - graphql
---

> Today we are releasing further improvements to the schema customization that [we've released in version 2.2.0](/blog/2019-03-18-releasing-new-schema-customization). You can use them with Gatsby 2.5.0.

Reviewer: ```suggestion
Today we are releasing further improvements to the schema customization capabilities [we released in version 2.2.0](/blog/2019-03-18-releasing-new-schema-customization). You can use them with Gatsby 2.5.0.
```

It is now possible to indicate to Gatsby, that you want to add a resolver to an explicitly defined fields. Use extensions like `@link` and `@dateformat` to add default arguments or/and resolvers to fields. In addition, when `@dontInfer` is set, Gatsby will no longer run inference for marked type, allowing one to improve performance for large data sets.

## Summary

> After about a month of testing schema customization both here and in pre-release we determined a couple of issues. The original aim of our schema customisation work was to remove uncertainty in user's schemas when their data changes.

Reviewer: ```suggestion
After about a month of testing schema customization both pre and post release, we determined a couple of issues. The original aim of our schema customisation work was to remove uncertainty in a user's schemas when their data changes.
```

Applies with `@infer` or `@infer(noDefaultResolvers: true)`.

Type gets all inferred fields added. Existing fields won't automatically get resolvers (use the provided resolver extensions).

#### No inference

Applies with `@dontInfer` or `@dontInfer(noDefaultResolvers: true)`.

Inference won't run at all. Existing fields won't automatically get resolvers (use resolver extensions).

#### No new fields with default resolvers (deprecated, removed in v3)

Applies with `@dontInfer(noDefaultResolvers: false)`

> Inference will run, but fields won't be added. If a type has defined fields of types `Date`, `File`, and any other node, and we inferred that they should have resolvers/args. These will still be added in Gatsby v2, but we will now warn and note the future deprecation.

Author: ```suggestion
Inference will run, but fields won't be added. If a type has defined fields of types `Date`, `File`, and any other node, and we inferred that they should have resolvers/args. These will still be added in Gatsby v2, but we will now warn and note the deprecation.
```
